Lib/test/test_capi/test_opt.py

```diff
@@ -1437,6 +1437,38 @@ def crash_addition():
 
         crash_addition()
 
+    def test_narrow_type_to_constant_str_empty(self):
+        def f(n):
+            trace = []
+            for i in range(n):
+                dummy = "aaa"
+                # Hopefully the optimizer can't guess what the value is.
+                # empty is always "", but we can only prove that it's a string:
+                empty = dummy[:0]
```

Comment:
I can easily see the optimizer turning `"aaa"[:0]` into `""`.
`empty` doesn't need to be a constant, we just need it to be mostly "", for profiling.
Use something like `empty = "a"[:(n % 1000) == 0]`

Reply:
Since we check the actual path taken as part of the test, we need the value is always `""`, not just mostly `""`. So maybe:

```py
false = i != TIER2_THRESHOLD
empty = "X"[:false]
```

The optimizer can't prove `false` is `False`, so it's good enough for our purposes.